File: 1.arrays/max-profit/max_profit.py
def max_profit(prices) -> int:
  if not prices:
    return 0
  minPrice = prices[0]
  maxPrice = 0
  for i in range(1, len(prices)):
    minPrice = min(prices[i], minPrice)
    maxPrice = max(prices[i],maxPrice)
  return maxPrice


arr = [100, 180, 260, 310, 40, 535, 695]
print(max_profit(arr))

File: 1.arrays/max-profit/maxProfit.js
function maxProfit(prices) {
  if (!prices) {
    return 0;
  }
  let maxPrice = 0;
  let minPrice = prices[0];
  for (let i = 1; i < prices.length; ++i) {
    minPrice = Math.min(minPrice, prices[i]);
    maxPrice = Math.max(maxPrice, prices[i]);
  }
  return maxPrice;
}

const arr = [100, 180, 260, 310, 40, 535, 695];
console.log(maxProfit(arr));
